backends/backend_libkcapi.c

```diff
@@ -1148,3 +1148,383 @@ static void kcapi_mac_backend(void)
+                logger(LOGGER_DEBUG, "ecdsa: public keyVer success\n");
+        }
+out:
+        kcapi_ecc_destroy(handle);
+        return ret;
+}
+
+static int ecdsa_sigver(struct ecdsa_sigver_data *data, flags_t parsed_flags)
+{
+        if(!data)
+        {
+                logger(LOGGER_DEBUG,"ecdsa: ecdsa_sigver_data is empty, returning -EINVAL...\n");
+                return -EINVAL;
+        }
+
+        struct kcapi_handle *handle = NULL, *hash_handle = NULL;
+        int ret = 0, rc = 0;
+        size_t len = 0;
+        char *curve_str = NULL;
+        (void) parsed_flags;
+
+        if((data->cipher & ACVP_CURVEMASK) == ACVP_NISTP384)
+        {
+                curve_str = "ecdsa-nist-p384";
+        }
+        else if((data->cipher & ACVP_CURVEMASK) == ACVP_NISTP256)
+        {
+                curve_str = "ecdsa-nist-p256";
+        }
+        else
+        {
+                logger(LOGGER_DEBUG, "ecdsa: curve not FIPS supported.\n");
+                return -EINVAL;
+        }
+
+        logger_binary(LOGGER_DEBUG, data->msg.buf, data->msg.len, "ecdsa: msg: ");
+
+        if(!data->component)
+        {
+                /* We need to generate the msg hash */
+                struct buffer msg = data->msg;
+                data->msg.buf = NULL;
+                data->msg.len = 0;
+                char cipher[CIPHERMAXNAME];
+
+                if(sha_mac_cipher(cipher, data->cipher & ACVP_HASHMASK, &len))
+                {
+                        return -EINVAL;
+                }
+
+                if (kcapi_md_init(&hash_handle, cipher, 0))
+                {
+                        logger(LOGGER_DEBUG, "ecdsa: allocation of hash %s failed\n", cipher);
+                        return 1;
+                }
+
+                CKINT_LOG(alloc_buf(len, &data->msg), "ecdsa: msg hash buffer could not be allocated\n");
+                rc = kcapi_md_digest(hash_handle, msg.buf, msg.len, data->msg.buf, data->msg.len);
+                if (rc < 0)
+                {
+                        logger(LOGGER_DEBUG, "ecdsa: message digest generation failed\n");
+                        kcapi_md_destroy(hash_handle);
+                        return 1;
+                }
+
+                logger_binary(LOGGER_DEBUG, data->msg.buf, data->msg.len, "ecdsa: msg hash: ");
+                kcapi_md_destroy(hash_handle);
+                if(msg.buf)
+                        free_buf(&msg);
+        }
+
+        if (kcapi_akcipher_init(&handle, curve_str, 0))
+        {
+                logger(LOGGER_DEBUG, "ecdsa: allocation of %s cipher failed\n", curve_str);
+                return -EFAULT;
+        }
+
+        /* Encoding and setting public key */
+        struct buffer pk;
+        unsigned char *ptr;
+        pk.len = 0;
+        pk.buf = NULL;
+
+        CKINT_LOG(alloc_buf(data->Qx.len + data->Qy.len + 1, &pk), "ecdsa: public key buffer could not be allocated\n");
+        ptr = (&pk)->buf;
+
+        ptr[0] = 0x04;
```

Comment:
why is this hardcoded to 0x04? Can you make it a macro with self explanatory name?